ETL/test/handle.cpp

```diff
@@ -327,6 +338,208 @@ int rhandle_general_use_test(void)
 		printf(__FILE__":%d: On clear, instance count=%d, should be zero.\n",__LINE__,my_test_obj::instance_count);
 		return 1;
 	}
+    obj_handle obj1;
+	if(!obj1.empty()){ //! default handle constructor should return an empty pointer
+        printf("FAILED!\n");
+        printf(__FILE__":%d: on create an empty handle, the number of instances=%d must be zero.\n",__LINE__,obj.count());
+        return 1;
+	}
+    robj_handle_old robj1(new my_test_obj(rand()));
+	if(robj1.count()!=1){
+        printf("FAILED!\n");
+        printf(__FILE__":%d: on create rhandle instance count=%d, should be 1.\n",__LINE__,obj1.count());
+        return 1;
+	}
+	obj1 = robj1;
+    if(obj1.count()!=2){ //! on copy "handle" , counter should increase by 1
+        printf("FAILED!\n");
+        printf(__FILE__":%d: On copy, handle count=%d, should be 2.\n",__LINE__,obj1.count());
+        return 1;
+    }
+	robj1.detach();
+    //! After detaching, the object should not be destroyed if it is referenced by other pointers
+    //! and the counter should be one less than its previous value.
+    if(obj1.count()!=1){
+        printf("FAILED!\n");
+        printf(__FILE__":%d: On copy, handle count=%d, should be 1.\n",__LINE__,obj1.count());
+        return 1;
+    }
+    obj1.detach();
+
+    my_test_obj::instance_count=0;
+    { robj_handle_old robj = obj_handle(etl::loose_handle<my_test_obj>()); }
+    if(my_test_obj::instance_count){
+        printf("FAILED!\n");
+        printf(__FILE__":%d: On clear, instance count=%d, should be zero.\n",__LINE__,my_test_obj::instance_count);
+    }
+
+	printf("PASSED\n");
+
+	return 0;
+}
+
+int rhandle_new_general_use_test(void)
```

Comment:
Yes, it looks much better that way.

Next step. Since these functions (`rhandle_new_general_use_test`/`rhandle_old_general_use_test`) should be identical (except for the type of pointer used), I suggest rewriting them using a template.

In this case, you can be completely sure that the tests are identical.